src/modules/Dashboard/features/RespondentData/RespondentDataReview/RespondentDataReview.tsx

```diff
@@ -143,12 +143,14 @@ export const RespondentDataReview = () => {
     const datesResult = data?.result;
     if (!datesResult) return;
 
+    // Map each date string to a Date object, appending 'T00:00:00' to remove timezone offset
     const submitDates = datesResult.dates.map((date: string) => new Date(`${date}T00:00:00`));
+
     setResponseDates(submitDates);
   });
 
   const handleGetSubmitDates = (date: Date) => {
-    if (!appletId || !subjectId) return;
+    if (!appletId || !subjectId || !selectedActivity?.id) return;
```

Comment:
We don't actually need activity ID to be defined as it's an optional parameter to `getAppletSubmitDateList` (but open to your thoughts).
```suggestion
    if (!appletId || !subjectId) return;
```

Reply:
This was necessary to make sure the method in the api is being called with a non-empty value for `activityOrFlowId` @farmerpaul

Reply:
I'm not sure we should be restricting the `handleGetSubmitDates` to only instances of this screen where an activity has been selected. The `RespondentDataReview` screen is accessible in the app from several routes, including a route that doesn't include either an activity ID or flow ID as a parameter:

https://github.com/ChildMindInstitute/mindlogger-admin/blob/c1ba7658fb4e4cb934bd84dfe0041d2be4e121a8/src/modules/Dashboard/routes/routes.tsx#L136-L147

https://github.com/ChildMindInstitute/mindlogger-admin/blob/a55547a273aa2798fa9b264f2481b71cbb860d40/src/resources/pages.json#L27

So it's still possible to invoke this component where activity ID or activity flow ID have not been defined. I was able to navigate to this URL after checking out your branch:

`https://localhost:3000/dashboard/8e9bd4f9-7ea3-4cfd-b6b1-040108774370/participants/83b7ff36-2175-4fd0-80c7-bc9f1d08ddee/dataviz/responses`

And because no default activity was selected, the `handleGetSubmitDates` callback was bypassed due to this condition, and resulted in a datepicker showing all dates as being available:

![CleanShot 2024-12-11 at 16 47 16@2x](https://github.com/user-attachments/assets/6a65adb4-f856-448e-927e-0287a5a61fb6)

Now, whether the app is even supposed to still be calling this route is a separate question (I think it's deprecated), one which I think I need to bring up with product. But I think it's still accessible from a few spots, so we should probably make sure not to affect the datepicker when those routes are still being called.